syscalls_bsd.go

```diff
@@ -0,0 +1,45 @@
+// +build openbsd freebsd
+
+package water
+
+import (
+	"os"
+)
+
+const bsdTUNSIFINFO = (0x80000000) | ((4 & 0x1fff) << 16) | uint32(byte('t'))<<8 | 91
+const bsdTUNSIFMODE = (0x80000000) | ((4 & 0x1fff) << 16) | uint32(byte('t'))<<8 | 94
+
+type tuninfo struct {
+	BaudRate int
+	MTU      int16
+	Type     uint8
+	Dummy    uint8
+}
```

Comment:
Do these fields need to be exported?

Reply:
Actually it seems lines above this are not used at all. I was imagining you needed to make some syscalls to setup the tun/tap device. In your use case, do you call external commands to set them up? I think if that's the preferred use case, you wouldn't need this package at all. You can just use `os.OpenFile` directly.

I see there the second commit removed a ioctl function. Did you attempt to implement it first and removed it? I think it'd be nice to have syscalls handled in Go!

Reply:
I do appear to have left this definition in by accident, as this would be handled elsewhere in my code (with the help of `FD()` above). 

I did attempt to implement an `ioctl` function although then realised that other platforms do not necessarily have this implemented either, hence removal.

Reply:
> then realised that other platforms do not necessarily have this implemented either, hence removal.

I think all three platforms that this package has support for already handle syscalls properly. They might be structured in different ways because of different underlying designs of the OS APIs, but the goal is a user of the package doesn't need to make syscalls before or after creating the interface using this package.

I imagine the OpenBSD version could be similar to `syscalls_darwin.go` and `syscalls_linux.go`.